yt_dlp/extractor/tiktok.py

```diff
@@ -220,7 +123,7 @@ def extract_addr(addr, add_meta={}):
                 'vcodec': 'h264',
                 'width': video_info.get('width'),
                 'height': video_info.get('height'),
-                'source_preference': -2 if video_info.get('has_watermark') else -1,
+                'preference': -2 if video_info.get('has_watermark') else -1,
```

Comment:
```
                    * preference Order number of this format. If this field is
                                 present and not None, the formats get sorted
                                 by this field, regardless of all other values.
```
Are you sure this is what you want? The priority of `preference` cannot be adjusted using `_sort_formats`. I believe this should in fact be `source_preference` here